modules/trusted-firmware-m/Kconfig

```diff
@@ -1,377 +1,12 @@
-	  Setting this option will cause '-DTFM_PARTITION_AUDIT_LOG'
-	  to be passed to the TF-M build system. Look at 'config_default.cmake'
-	  in the trusted-firmware-m repository for details regarding this
-	  parameter. Any dependencies between the various TFM_PARTITION_*
-	  options are handled by the build system in the trusted-firmware-m
-	  repository.
-	  Note: the Audit Log service does not implement the IPC model
-	  interface so it may not be enabled together with IPC option.
-
-config TFM_IPC
-	bool "IPC" if TFM_PROFILE_TYPE_NOT_SET
-	default y if (TFM_PROFILE_TYPE_MEDIUM || TFM_PROFILE_TYPE_LARGE)
-	help
-	  When enabled, this option signifies that the TF-M build supports
-	  the PSA API (IPC mode) instead of the secure library mode. When
-	  TF-M Profile option is supplied, do not allow manual setting of
-	  the IPC mode, as it is determined by the profile setting.
-
-config TFM_REGRESSION_S
-	bool "TF-M Secure Regression tests"
-	help
-	  When enabled, this option signifies that the TF-M build includes
-	  the Secure domain regression tests.
-
-config TFM_REGRESSION_NS
-	bool "Use the TF-M Non-Secure Regression test application"
-	help
-	  When this is enabled, the Zephyr application as a whole will be
-	  replaced with the TF-M Non-Secure Regression test application.
-
-choice TFM_PSA_TEST
-	prompt "Enable a PSA test suite"
-	default TFM_PSA_TEST_NONE
-
-config TFM_PSA_TEST_CRYPTO
-	bool "Crypto tests"
-	depends on MAIN_STACK_SIZE >= 4096
-	help
-	  Enable the PSA Crypto test suite.
-
-config TFM_PSA_TEST_PROTECTED_STORAGE
-	bool "Storage tests"
-	help
-	  Enable the PSA Protected Storage test suite.
-
-config TFM_PSA_TEST_INTERNAL_TRUSTED_STORAGE
-	bool "Internal Trusted Storage tests"
-	help
-	  Enable the PSA Internal Trusted Storage test suite.
-
-config TFM_PSA_TEST_STORAGE
-	bool "Storage tests"
-	help
-	  Enable the PSA Storage test suite. This is a combination of the
-	  protected storage and internal trusted storage tests.
-
-config TFM_PSA_TEST_INITIAL_ATTESTATION
-	bool "Initial attestation tests"
-	depends on MAIN_STACK_SIZE >= 4096
-	help
-	  Enable the PSA Initial Attestation test suite.
-
-config TFM_PSA_TEST_NONE
-	bool "No PSA test suite"
-
-endchoice
-
-if TFM_BL2
-
-config ROM_START_OFFSET
-	hex "ROM Start Offset accounting for BL2 Header in the NS image"
-	default 0x400
-	help
-	  By default BL2 header size in TF-M is 0x400. ROM_START_OFFSET
-	  needs to be updated if TF-M switches to use a different header
-	  size for BL2.
-
-endif # !TFM_BL2
-
-# Option to instruct flashing a merged binary consisting of BL2 (optionally),
-# TF-M (Secure), and application (Non-Secure).
-config TFM_FLASH_MERGED_BINARY
-	bool
-	help
-		This option instructs west flash to program the
-		combined (merged) binary consisting of the TF-M
-		Secure firmware image, optionally, the BL2 image
-		(if building with TFM_BL2 is enabled), and the
-		Non-Secure application firmware.
-
-endif # BUILD_WITH_TFM
+rsource "Kconfig.tfm.partitions"
```

Comment:
Both these files are "covered" internally by `if BUILD_WITH_TFM` - is it still necessary to have that check here?

Reply:
Removed, @hakonfam , thanks